test/common_utils.py

```diff
@@ -55,24 +54,14 @@ def random_float_tensor(seed, size, a=22695477, c=1, m=2 ** 32):
     return torch.tensor(arr).float().view(size) / m
 
 
-@contextmanager
-def AudioBackendScope(new_backend):
-    previous_backend = torchaudio.get_audio_backend()
-    try:
-        torchaudio.set_audio_backend(new_backend)
-        yield
-    finally:
-        torchaudio.set_audio_backend(previous_backend)
-
-
 def filter_backends_with_mp3(backends):
     # Filter out backends that do not support mp3
     test_filepath = get_asset_path('steam-train-whistle-daniel_simon.mp3')
 
     def supports_mp3(backend):
+        torchaudio.set_audio_backend(backend)
```

Comment:
nit: at the end of `filter_backends_with_mp3`, the backend will have changed. i see this done in many places. A context manager would have been nice to revert the change of backend at the end, but adds complexity for little gain.

Reply:
It is idividual unit test's responsibility to set the required condition. So there is not much point of reverting the backend.